dataloader.go

```diff
@@ -20,18 +20,47 @@ import (
 // different access permissions and consider creating a new instance per
 // web request.
 type Interface interface {
-	Load(context.Context, interface{}) Thunk
-	LoadMany(context.Context, []interface{}) ThunkMany
-	Clear(context.Context, string) Interface
+	Load(context.Context, Keyer) Thunk
+	LoadMany(context.Context, KeyList) ThunkMany
+	Clear(context.Context, Keyer) Interface
 	ClearAll() Interface
-	Prime(ctx context.Context, key interface{}, value interface{}) Interface
+	Prime(ctx context.Context, key Keyer, value interface{}) Interface
+}
+
+// StringKey implements the Key interface for a string
+type StringKey string
+
+// Key is an identity method. Used to implement Key inteface
+func (k StringKey) Key() string { return string(k) }
+
+// String is an identity method. Used to implement Key Value
+func (k StringKey) Value() interface{} { return k }
+
+// Keyer is the interface that all keys need to implement
+type Keyer interface {
```

Comment:
I know the `X-er` pattern is common in go but it feels a little weird here, especially since the interface specifies more methods than just `Key()`. 

Do you think we can get away with calling the interface `Key`? This would help improve consistency with the `KeyList` type.

```diff
- type Keyer interface
+ type Key interface

- type KeyList []Keyer
+ type KeyList []Key
```